### PR TITLE
Feature/refactor code

### PR DESCRIPTION
**Pull request by Eukst**
Refactor Code 
Improved UI
Added Pagination Feature
added more views to user , admin

### CODE REVIEW

**Review by harsh-ind — APPROVED** [review]

File: app/controllers/apply_jobs_controller.rb [hunk 1]
```diff
@@ -27,11 +28,14 @@ def create
     end
   end
 
+  # delete apply_jobs/:id
   def destroy
+    p params[:user_id].to_i
+    p params[:job_id].to_i
```
Comment (mitaligaur) [inline app/controllers/apply_jobs_controller.rb]: Remove this

File: app/controllers/jobs_controller.rb [hunk 2]
```diff
@@ -11,30 +12,27 @@ def index
   # GET /jobs/1 or /jobs/1.json
   def show
     if user_signed_in? && (current_user.user_role.include? 'Employer')
-      #@jobs = Job.where(user_id: current_user.id)
-      #jobs_ids = @job.id
-      #p jobs_ids
-      #apply_user_ids = ApplyJob.where('job_id IN (?)', jobs_ids).pluck(:user_id).uniq
-      @user_apply = @job.users   #.where(id: apply_user_ids)
+
+      @user_apply = @job.users # .where(id: apply_user_ids)
```
Comment (mitaligaur) [inline app/controllers/jobs_controller.rb]: Remove commented part

File: app/controllers/static_pages_controller.rb [hunk 3]
```diff
@@ -2,13 +2,24 @@
...
   def home
     if user_signed_in? && (current_user.user_role.include? 'Employer')
       @jobs = Job.where(user_id: current_user.id)
+      @jobs = Kaminari.paginate_array(@jobs).page(params[:page]).per(7)
       jobs_ids = Job.where(user_id: current_user.id).pluck(:id)
       apply_user_ids = ApplyJob.where('job_id IN (?)', jobs_ids).pluck(:user_id).uniq
       @user_apply = User.where(id: apply_user_ids)
+    end
+
+    if user_signed_in? && (current_user.user_role.include? 'Job Seeker') && (current_user.admin == false)
```
Comment (mitaligaur) [inline app/controllers/static_pages_controller.rb]: Use !current_user.admin


Also update in related code
Check in other files too

File: app/helpers/apply_job_helper.rb [hunk 4]
```diff
@@ -1,11 +1,24 @@
...
-        else
-            return true
-        end
+  # Return a user already applied for job or not
+  def applied?(user_id)
+    @apply_job = ApplyJob.find_by(user_id: user_id, job_id: params[:id].to_i)
+    if @apply_job.nil?
+      false
+    else
+      true
```
Comment (mitaligaur) [inline app/helpers/apply_job_helper.rb]: Change this into ---



return false if @apply_job.nil?

true

File: app/helpers/apply_job_helper.rb [hunk 5]
```diff
@@ -1,11 +1,24 @@
...
+  end
 
+  # Return a user already applied for job or not (takes two parameter)
+  def applied_to_job?(user_id, job_id)
+    @apply_job = ApplyJob.find_by(user_id: user_id, job_id: job_id)
+    if @apply_job.nil?
+      false
+    else
+      true
+    end
```
Comment (mitaligaur) [inline app/helpers/apply_job_helper.rb]: Modify this too like mentioned above

**Review by mitaligaur — CHANGES_REQUESTED** [review]

**Review by Sharee-ongraph — APPROVED** [review]

File: app/controllers/apply_jobs_controller.rb (hunk 1 above)
Comment (Eukst) [reply app/controllers/apply_jobs_controller.rb]: done

File: app/controllers/jobs_controller.rb (hunk 2 above)
Comment (Eukst) [reply app/controllers/jobs_controller.rb]: done

File: app/helpers/apply_job_helper.rb (hunk 4 above)
Comment (Eukst) [reply app/helpers/apply_job_helper.rb]: done

File: app/helpers/apply_job_helper.rb (hunk 5 above)
Comment (Eukst) [reply app/helpers/apply_job_helper.rb]: done

File: app/controllers/static_pages_controller.rb (hunk 3 above)
Comment (Eukst) [reply app/controllers/static_pages_controller.rb]: done

File: app/helpers/apply_job_helper.rb [hunk 6]
```diff
@@ -1,11 +1,23 @@
...
-    end
-end
+  # Return a user already applied for job or not
+  def applied?(user_id)
+    @apply_job = ApplyJob.find_by(user_id: user_id, job_id: params[:id].to_i)
+    return false if @apply_job.nil?
+
+    true
 
+
```
Comment (mitaligaur) [inline app/helpers/apply_job_helper.rb]: Remove blank line

File: app/helpers/apply_job_helper.rb [hunk 7]
```diff
@@ -1,11 +1,23 @@
...
+ 
+  end
+
+  # Return a user already applied for job or not (takes two parameter)
+  def applied_to_job?(user_id, job_id)
+    @apply_job = ApplyJob.find_by(user_id: user_id, job_id: job_id)
+    return false if @apply_job.nil?
+
+    true
+
```
Comment (mitaligaur) [inline app/helpers/apply_job_helper.rb]: Remove blank line

**Review by mitaligaur — CHANGES_REQUESTED** [review]

**Review by mitaligaur — APPROVED** [review]

**Pull request merged**